dashboard/src/features/accounts/components/AccountUpdate.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import { BaseUpdate, FormContainer, FormSection, JsonField, NotFound } from 'features/shared/components'
+import { reduxForm } from 'redux-form'
+
+class Form extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.submitWithErrors = this.submitWithErrors.bind(this)
+
+    this.state = {}
+  }
+
+  submitWithErrors(data) {
+    return new Promise((resolve, reject) => {
+      this.props.submitForm(data, this.props.item.id)
+        .catch((err) => reject({_error: err}))
+    })
+  }
+
+  componentDidMount() {
+    this.props.fetchItem(this.props.params.id).then(resp => {
+      if (resp.items.length == 0) {
+        this.setState({notFound: true})
+      }
+    })
+  }
+
+  renderIfFound(view) {
+    if (this.state.notFound) {
+      return(<NotFound />)
+    } else if (view) {
+      return(view)
+    } else {
+      return(<div>Loading...</div>)
+    }
+  }
+
+  render() {
+    const item = this.props.item
```

Comment:
I think this may have been copied from another component, but you may be able to get away with one less function if your `render` looked like this:

```
render() {
  if (this.state.notFound) {
    return <NotFound />
  }

  if (!this.props.item) {
    return <div>Loading...</div>
  }

  // stuff for rendering the form
}
```